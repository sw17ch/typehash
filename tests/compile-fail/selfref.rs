use typehash::TypeHash;


#[derive(TypeHash)]
struct SelfRef {
    _ref: *const SelfRef,
}
