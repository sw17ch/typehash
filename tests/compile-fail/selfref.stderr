error: The type referred to by field '_ref' is not supported: recursive type detected.
 --> $DIR/selfref.rs:4:10
  |
4 | #[derive(TypeHash)]
  |          ^^^^^^^^
  |
  = note: this error originates in the derive macro `TypeHash` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0601]: `main` function not found in crate `$CRATE`
 --> $DIR/selfref.rs:1:1
  |
1 | / use typehash::TypeHash;
2 | |
3 | |
4 | | #[derive(TypeHash)]
5 | | struct SelfRef {
6 | |     _ref: *const SelfRef,
7 | | }
  | |_^ consider adding a `main` function to `$DIR/tests\compile-fail\selfref.rs`
